includes/User.php

```diff
@@ -2169,6 +2167,8 @@ public function invalidateCache() {
 					__METHOD__ );
 			}
 
+			$dbw->commit();
```

Comment:
I assume that we want to make user preferences / attributes / whatever service be the **only** source of users data. Shouldn't we instead of this fix implement a method in a service that will return "user touched" from services point of view - this will be used to invalidate MW cache (I believe we still need it).